packages/flutter/lib/src/widgets/shortcuts.dart

```diff
@@ -1058,3 +1062,287 @@ class CallbackShortcuts extends StatelessWidget {
+  ///
+  /// {@macro flutter.widgets.ProxyWidget.child}
+  final Widget child;
+
+  /// Returns the [ShortcutRegistry] that belongs to the [ShortcutRegistrar]
+  /// which most tightly encloses the given [BuildContext].
+  ///
+  /// If no [ShortcutRegistrar] widget encloses the context given, `of` will
+  /// throw an exception in debug mode.
+  ///
+  /// The dependencies of [ShortcutRegistrar] will have their
+  /// [State.didChangeDependencies] called whenever the shortcuts have changed.
+  ///
+  /// See also:
+  ///
+  ///  * [maybeOf], which is similar to this function, but will return null if
+  ///    it doesn't find a [ShortcutRegistrar] ancestor.
+  static ShortcutRegistry of(BuildContext context) {
+    assert(context != null);
+    final _ShortcutRegistrarMarker? inherited =
+        context.dependOnInheritedWidgetOfExactType<_ShortcutRegistrarMarker>();
+    assert(() {
+      if (inherited == null) {
+        throw FlutterError(
+          'Unable to find a $ShortcutRegistrar widget in the context.\n'
+          '$ShortcutRegistrar.of() was called with a context that does not contain a '
+          '$ShortcutRegistrar widget.\n'
+          'No $ShortcutRegistrar ancestor could be found starting from the context that was '
+          'passed to $ShortcutRegistrar.of().\n'
+          'The context used was:\n'
+          '  $context',
+        );
+      }
+      return true;
+    }());
+    return inherited!.registry;
+  }
+
+  /// Returns [ShortcutRegistry] of the [ShortcutRegistrar] that
+  /// most tightly encloses the given [BuildContext].
+  ///
+  /// If no [ShortcutRegistrar] widget encloses the given context,
+  /// `maybeOf` will return null.
+  ///
+  /// The dependencies of [ShortcutRegistrar] will have their
+  /// [State.didChangeDependencies] called whenever the shortcuts have changed.
+  ///
+  /// See also:
+  ///
+  ///  * [of], which is similar to this function, but returns a non-nullable
+  ///    result, and will throw an exception if it doesn't find a
+  ///    [ShortcutRegistrar] ancestor.
+  static ShortcutRegistry? maybeOf(BuildContext context) {
+    assert(context != null);
+    final _ShortcutRegistrarMarker? inherited =
+        context.dependOnInheritedWidgetOfExactType<_ShortcutRegistrarMarker>();
+    return inherited?.registry;
+  }
+
+  @override
+  State<ShortcutRegistrar> createState() => _ShortcutRegistrarState();
+}
+
+class _ShortcutRegistrarState extends State<ShortcutRegistrar> {
+  late ShortcutRegistry registry;
+  bool updateScheduled = false;
+
+  @override
+  void initState() {
+    super.initState();
+    registry = ShortcutRegistry();
+    registry.addListener(handleRegistryChange);
+  }
+
+  @override
+  void dispose() {
+    registry.removeListener(handleRegistryChange);
+    registry.dispose();
+    super.dispose();
+  }
+
+  void handleRegistryChange() {
+    if (!updateScheduled) {
+      updateScheduled = true;
+      SchedulerBinding.instance.addPostFrameCallback((Duration _) {
```

Comment:
OK, I've modified the registry to not update its dependencies when the shortcuts change (only when the registry instance itself changes), and I was able to make the `ShortcutRegistry` just be a specialized `ShortcutManager` that allows registration.

In addition, I modified the `ShortcutManager` so that it also doesn't notify when the shortcuts change, since that is likely causing unnecessary updates.  If someone needs to update their UI in response to shortcut changes, they can listen to the manager directly.